docs/docs/09-rfcs/2023-03-08-wing-preview-environment.md

```diff
@@ -0,0 +1,249 @@
+
+## Requirements
+
+Wing Cloud users can build and deploy Wing applications to an ephemeral (simulated) environment and use the Wing Console UI to interact with it.
+Automatically run tests on the preview environment once created and provide a UI for viewing the test results.
+Allow developers to collaborate with their teammates using a shared environment.
+Seamlessly integrating with GitHub.
+
+### Sign Up to Wing Cloud
+
+1. Goto https://wing.cloud and click on sign up.
+2. sign up to Wing Cloud using GitHub authentication.
+
+### Sign In to Wing Cloud
+
+1. Goto https://wing.cloud and click on sign in.
+2. If a developer was already signed in before, he will be automatically signed in to Wing Cloud.
+
+### Authentication and Authorization
+
+1. Wing Cloud Preview Environments will use GitHub for authentication and authorization.
+2. Developers will be able to install Wing Cloud Preview Environments on a repository only if they have the GitHub permissions.
+3. Developers will be able to create a PR with a preview environment only if they have the necessary GitHub permissions to the repository **and** their GitHub user is signed up to Wing Cloud.
+4. Everyone can view a preview environment if they have the url.
+
+### Wing Cloud Preview Environments Developer Experience
+
+A developer using winglang has the capability to integrate Wing Cloud Preview Environments with their repository.
+
+#### Installation
+
+In order to use Wing Cloud Preview Environments, you should sign in to Wing Cloud.
+The installation process is straightforward using GitHub application:
+
+1. Goto https://wing.cloud and log in
+2. Goto https://wing.cloud/preview and Click on "Deploy with Wing Cloud"
+3. Complete GitHub application authentication.
+4. Grant repository access permission.
+5. You are now ready to use Wing Cloud Preview Environments.
+6. A welcome email will be sent to you with additional information and links to documentation.
+
+#### Preview Environment for production branch
+
+For the production branch, there is a single preview environment that will constantly be up-to-date with the latest code committed.
+A link to the preview environment is available in the repository main page ("About" section).
+Production branch environment url structure is: `https://<gh-repository>-<gh-branch>-<entry-point>.wing.cloud.app`
+
+#### Preview Environments for Pull Requests
+
+Upon each creation of a pull request an automatic comment will be added to the PR and will guide the developer to dedicated preview environments.
+provide a streamlined process for updating a pull request preview environment, each PR code changes will redeploy the preview environment and also will:
+
+1. provide a real-time build and deployment status updates
+2. each entry point will have a seperated preview environment
+3. ensure a consistent and unique url for each preview environment. The url structure: `https://<gh-repository>-<gh-branch>-<entry-point>-<pr-number>.wing.cloud.app`
```

Comment:
@ainvoner Instead of having an additional subdomain with the PR number, I'd use a random id mixed with repo information such as the repo name and/or org, like Vercel does. It's easier for us to generate a unique ID this way, and also generate an alias for it using `https://<gh-repository>-<gh-branch>-<entry-point>.wingcloud.app`, if available.